packages/ui/src/modal/index.ts

```diff
@@ -0,0 +1,53 @@
+import Modal from "./Modal.svelte"
+import type {
+  ConnectedStarknetWindowObject,
+  StarknetWindowObject,
+  WalletProvider,
+} from "get-starknet-core"
+
+export interface WalletProviderWithStoreVersion
+  extends Omit<WalletProvider, "downloads"> {
+  download: string
+}
+
+export default async function show({
+  discoveryWallets,
+  installedWallets,
+  lastWallet,
+  defaultWallet,
+  preAuthorizedWallets,
+  enable,
+  modalOptions,
+}: {
+  lastWallet?: StarknetWindowObject
+  defaultWallet?: StarknetWindowObject
+  installedWallets?: StarknetWindowObject[]
+  preAuthorizedWallets?: StarknetWindowObject[]
+  discoveryWallets?: WalletProviderWithStoreVersion[]
+  enable?: (
+    wallet: StarknetWindowObject | null,
+  ) => Promise<ConnectedStarknetWindowObject | null>
```

Comment:
we'll need `| undefined` for backward compatibility (we can have both `| null | undefined`)